pkg/aws_keyhub/aws.go

```diff
@@ -0,0 +1,118 @@
+package aws_keyhub
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/ini.v1"
+	"os"
+	"strings"
+)
+
+func StsAssumeRoleWithSAML(principalArn string, roleArn string, samlAssertion string) *sts.AssumeRoleWithSAMLOutput {
+	config := getAwsKeyHubConfig()
+	input := &sts.AssumeRoleWithSAMLInput{
+		DurationSeconds: aws.Int64(config.Aws.AssumeDuration),
+		PrincipalArn:    aws.String(principalArn),
+		RoleArn:         aws.String(roleArn),
+		SAMLAssertion:   aws.String(samlAssertion),
+	}
+
+	newSession, _ := session.NewSession()
+	svc := sts.New(newSession)
+	result, err := svc.AssumeRoleWithSAML(input)
+
+	if err != nil {
+		logrus.Fatal("AWS STS AssumeRoleWithSAML failed:", err)
+	}
+	logrus.Debugln("AWS STS AssumeRoleWithSAML result:", result)
+	return result
+}
+
+func VerifyIfLoginWasSuccessful(roleArn string) {
+
+	credentialsFromFile := credentials.NewSharedCredentials(getCredentialFilePath(), "keyhub")
+	config := &aws.Config{Credentials: credentialsFromFile}
+	newSession, _ := session.NewSession(config)
+	svc := sts.New(newSession)
+
+	input := &sts.GetCallerIdentityInput{}
+	result, err := svc.GetCallerIdentity(input)
+	if err != nil {
+		logrus.Fatal("Failed to get caller identity:", err)
+	}
+	logrus.Debugln("AWS STS GetCallerIdentity result:", result)
+
+	retrievedRoleArn := *result.Arn
+	// STS assumed role arn differs from IAM role arn
+	// IAM arn:aws:iam::123456789000:role/example-role
+	// STS arn:aws:sts::123456789000:assumed-role/example-role/example-username
```

Comment:
Persoonlijk ben ik ben niet zo een fan van commentaar in code omdat de code beschrijvend moet zijn. Maar dit is wel echt een goed voorbeeld wanneer commentaar wel veel toegevoegde waarde heeft. Wat de code naar mijn mening nog iets beschrijvender zou maken, is het vervangen van de replace door een methode zoals ReplaceStsRoleWithIamRole. Maar deze oplossing lijkt me ook prima.